### PR TITLE
[Linux] Fix power_operation_scripts failures

### PR DESCRIPTION
**Pull request by ZouYuhua**
Problem
-------------
power_operation_scripts hit intermittent failure for many guestOS such as Ubuntu and rhcos.

Root Cause:
-------------
Need to wait several seconds for wait configuration sync after modify etc/vmware-tools/tools.conf 

Test result:
Passed for Ubuntu 24.04 Live Server  and  Desktop, RHCOS.

### CODE REVIEW

File: linux/power_operation_scripts/set_power_script.yml [hunk 1]
```diff
@@ -48,28 +48,32 @@
     fail_msg: "Failed to find the {{ script_file_in_conf_file }} in tool.conf for {{ power_script_op }} script."
     success_msg: "Successfully found the {{ script_file_in_conf_file }} in tool.conf for {{ power_script_op }} script."
 
-- name: "Shutdown and then power on guest OS"
+- name: "Pause 10 seconds to wait configure take affected"
```
Comment (keirazhang) [inline linux/power_operation_scripts/set_power_script.yml:51]: `take affected` -> `take effect`

File: linux/power_operation_scripts/set_power_script.yml [hunk 2]
```diff
@@ -78,33 +82,64 @@
...
+        path: "{{ script_tag_path }}"
+      delegate_to: "{{ vm_guest_ip }}"
+      register: tag_path_exist_result
+      until:
+        - tag_path_exist_result.stat.exists is defined
+        - tag_path_exist_result.stat.exists
+      retries: 30
+      delay: 5
+
+    - name: "Check the serching result"
```
Comment (keirazhang) [inline linux/power_operation_scripts/set_power_script.yml:112]: There is a typo `serching`

File: linux/power_operation_scripts/set_power_script.yml [hunk 3]
```diff
@@ -48,28 +48,32 @@
     fail_msg: "Failed to find the {{ script_file_in_conf_file }} in tool.conf for {{ power_script_op }} script."
     success_msg: "Successfully found the {{ script_file_in_conf_file }} in tool.conf for {{ power_script_op }} script."
 
-- name: "Shutdown and then power on guest OS"
+- name: "Pause 10 seconds to wait configure take affected"
+  ansible.builtin.pause:
+    seconds: 10
+
+- name: "Shutdown and then Poweron guest OS for {{ power_script_type }} {{ power_script_op }}"
```
Comment (keirazhang) [inline linux/power_operation_scripts/set_power_script.yml:55]: `Poweron` -> `power on`

File: linux/power_operation_scripts/set_power_script.yml [hunk 4]
```diff
@@ -48,28 +48,32 @@
...
   when: power_script_op in ['power', 'shutdown']
   block:
-    - name: "Shutdown VM"
+    - name: "Shutdown VM for {{ power_script_type }} {{ power_script_op }}"
       include_tasks: ../../common/vm_set_power_state.yml
       vars:
         vm_power_state_set: "shutdown-guest"
 
-    - name: "Power on VM"
+    - name: "Poweron VM for {{ power_script_type }} {{ power_script_op }}"
```
Comment (keirazhang) [inline linux/power_operation_scripts/set_power_script.yml:63]: `Poweron` -> `Power on`

File: linux/power_operation_scripts/set_power_script.yml [hunk 5]
```diff
@@ -78,33 +82,64 @@
 - name: "Update inventory"
   include_tasks: ../../common/update_inventory.yml
 
-- name: "Check the script file running as expected"
-  ansible.builtin.stat:
-    path: "{{ script_tag_path }}"
+- name: "Wait for VMware Tools running"
+  include_tasks: ../../common/vm_wait_vmtools_status.yml
+  vars:
+    vm_wait_vmtools_running: true
```
Comment (keirazhang) [inline linux/power_operation_scripts/set_power_script.yml:88]: For VM with VMTools installed, update_inventory.yml should have waited tools running. Please check your test logs. If it already did, this task can be removed. Or you can move this task before `update_inventory.yml` to make sure `update_inventory.yml` getting ip from vmtools

File: linux/power_operation_scripts/set_power_script.yml [hunk 6]
```diff
@@ -78,33 +82,64 @@
...
+        fail_msg: "Failed to find {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}."
+        success_msg: "Successfully found the {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}."
 
 - name: "Check the log file"
   when: power_script_op != "suspend"
   block:
-    - name: "Search the log file"
-      ansible.builtin.shell: "grep '{{ script_file_path }}' /var/log/vmware-tools/*.log"
+    - name: "Search the log file {{ script_file_path }}"
+      ansible.builtin.shell: "grep {{ script_file_path | basename }} *.log"
```
Comment (keirazhang) [inline linux/power_operation_scripts/set_power_script.yml:124]: why do you replace file path with basename at here?

Comment (ZouYuhua) [reply linux/power_operation_scripts/set_power_script.yml]: change back to search file path.

File: linux/power_operation_scripts/set_power_script.yml [hunk 7]
```diff
@@ -78,33 +82,59 @@
...
 
     - name: "Check the serching result"
       ansible.builtin.assert:
         that:
           - result_search_str.rc is defined
           - result_search_str.rc == 0
-        fail_msg: "Failed to find the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_op }}."
-        success_msg: "Successfully found the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_op }}."
+        fail_msg: "Failed to find the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_type }} {{ power_script_op }}."
+        success_msg: "Successfully found the {{ script_file_path }} in {{ result_search_str.stdout }} for {{ power_script_type }} {{ power_script_op }}."
```
Comment (keirazhang) [inline linux/power_operation_scripts/set_power_script.yml:137]: I saw an example of the output of this task, and I think it is not friendly for reading the success message. 
```
TASK [Check the serching result] ***********************************************
task path: /home/worker/workspace/Ansible_RHCOS_MAIN-51/ansible-vsphere-gos-validation/linux/power_operation_scripts/set_power_script.yml:131
ok: [localhost] => {
    "changed": false,
    "msg": "Successfully found the /etc/vmware-tools/poweron-vm-default in vmsvc.log:[2024-10-14T05:53:25.854Z] [ message] [powerops] [936] Executing script: '/etc/vmware-tools/poweron-vm-default' for default power."
}
```

Comment (ZouYuhua) [reply linux/power_operation_scripts/set_power_script.yml]: Done to change it. Thanks

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**